lua/docs-view.lua

```diff
@@ -1,38 +1,35 @@
 local M = {}
-local config = {
+local default_cfg = {
   position = "right",
+  height = 10,
   width = 60,
 }
-
-M.setup = function(conf)
-  if conf.position then
-    config.position = conf.position
-  end
-
-  if conf.width then
-    config.width = conf.width
-  end
-end
-
 local buf, win, prev_win, autocmd
-M.toggle = function()
+
+local function toggle()
   if win and vim.api.nvim_win_is_valid(win) then
     vim.api.nvim_win_close(win, false)
     vim.api.nvim_del_autocmd(autocmd)
     buf, win, prev_win, autocmd = nil, nil, nil, nil
   else
     prev_win = vim.api.nvim_get_current_win()
 
-    if config.position == "left" then
+    if default_cfg.position == "bottom" then
+      vim.api.nvim_command("bel new")
+      default_cfg["width"] = vim.api.nvim_win_get_width(prev_win)
```

Comment:
I don't like the idea of mutation the `default_cfg`, I think using a variable local to the function would be better, something like this
```suggestion
	local width = default_cfg["width"];
    if default_cfg.position == "bottom" then
      vim.api.nvim_command("bel new")
      width = vim.api.nvim_win_get_width(prev_win)
```

Reply:
Okay! In that case, we would want to check if `width` and `height` are defined and use them for `math.ceil` conditionally, I think, like this:

```lua
vim.api.nvim_win_set_height(win, math.ceil(height or default_cfg.height))
vim.api.nvim_win_set_width(win, math.ceil(width or default_cfg.width)) 
```

Is this what you had in mind?

Reply:
with the introduction of `cfg` from the review comment above, it will be 
```lua
local width = cfg["width"]
local height = cfg["height"]

-- and then, height/width will surely be defined because cfg is a merge of default_cfg and use_cfg
vim.api.nvim_win_set_height(win, math.ceil(height))
vim.api.nvim_win_set_width(win, math.ceil(width)) 
```